### DISCUSSION TITLE
"Quick cast" spells in Items are always instant : cannot make any casted consumables/spells.

### DISCUSSION THREAD

**Discussion by Spectreye** [AscensionGameDev/Intersect-Engine#1496] [Ideas]
**Describe the bug**
When making an item with the spell attribute and linking a spell to it, the spell is given directly to you instead of being a one time cast.
That in fact, prevents us from doing anything other than instant spells with one time use.

And since you can already give a spell to a player with an "Event" Type spell, it would be nice to

**Version Encountered**
0.6.2.453

**To Reproduce**
1. Create a "Self" type spell "Heal" with a casting time of 1500 ms and whatever heal amount you want.
2. Create an item which use "Heal" as a spell. Do not activate quick cast.
3. Try to use the item in game.

**Expected behavior**
Items with the "Spell" Type attribute should be usable Without "quick casting" and giving spell would still be avaible via "Event" type items.
Or 
There should be one more choice : 
O : Quick cast ?
O : Destroy after use ?
**_O : Learn the spell ?_**

**Operating System:**
Windows

**Comment by Kamusvalenzuela** [comment]
That works well
One will learn the spell
and the quick cast will use instantly

**Comment by Spectreye** [comment]
> That works well
> One will learn the spell
> and the quick cast will use instantly

Edit : Ah i see what you told here. 

I've done test and i didn't see at first that it gave the spell, but i'm not sure it was intended to work like this cause there's already a mean to make the player "Get spell" defintively : 
If you want to make an item let you "Learn" a spell you can use the "Event" Type and make a common event that make you learn this spell.

But if you want to make a one time casted spell with an item you can't do it with the actual "quick cast" things since you have to make it an instant cast.

**Comment by jcsnider** [comment]
The intention is for:

Quick Cast - Casts the spell without it being taught (Like a Scroll in Skyrim)
Non Quick Cast (default) - Teaches the spell. (Like a Spell Tome in Skyrim)

I think this is working as intended.

**Comment by Kamusvalenzuela** [comment]
The choice 3 its more like an enhacement or new feature 
O : Learn the spell

**Comment by Spectreye** [comment]
> The intention is for:
> 
> Quick Cast - Casts the spell without it being taught (Like a Scroll in Skyrim)
> Non Quick Cast (default) - Teaches the spell. (Like a Spell Tome in Skyrim)
> 
> I think this is working as intended.

Actually the quick cast always "Cut" the casting time of the original spell, if it's intended that mean you can only do instant spells with it.

If it's the original intention i think it's a little restraining, why not let us the choice to have it instant or casted ?

**Comment by jcsnider** [comment]
So I think you all have worked this out.

Items with the Spell type either instantly teach the spell (put the spell into the spell book for future use) or instantly fire that spell. There is not currently an option for a single spell cast that also respects the spells cast timer/shows the casting bar.

This is working as intended, but other functionality/options could be added. As such I am marking this as an enhancement/feature request instead of a bug.
